<launch>
	<!-- vim: set ft=xml noet : -->


	<!-- image view params -->
	<arg name="front_cam_enable" default="false" />
	<arg name="down_cam_enable" default="true" />
	<arg name="videodev_front" default="2" />
	<arg name="videodev_down" default="0" />

	
<!--	<env name="ROSCONSOLE_CONFIG_FILE" value="$(find env_percept)/config/rosconsole.config" /> -->
<!--	<env name="ROSCONSOLE_FORMAT" value="[${severity}] [${time:%H:%M:%S}]: ${message}" /> -->

	<node if="$(arg down_cam_enable)" pkg="usb_cam" type="usb_cam_node" name="down_cam" output="screen" >
	  <param name="video_device" value="/dev/video$(arg videodev_down)" />
	  <param name="image_width" value="640" />
	  <param name="image_height" value="480" />
	  <param name="pixel_format" value="mjpeg" />
	  <param name="camera_frame_id" value="down_cam" />
	  <!-- <param name="camera_frame_id" value="logitech/fisheye" /> -->
	  <param name="io_method" value="mmap" />
	  <param name="framerate" value="30" />
	  <param name="autoexposure" value="false" />
	  <param name="camera_info_url" value="file://$(find env_percept)/config/down_cam.yaml" />
	  <param name="camera_name" value="down_cam" />
	  <!-- <param name="camera_name" value="logitech/fisheye" /> -->
	</node>


	<node if="$(arg front_cam_enable)" pkg="usb_cam" type="usb_cam_node" name="front_cam" output="screen" >
	  <param name="video_device" value="/dev/video$(arg videodev_front)" />
	  <param name="image_width" value="640" />
	  <param name="image_height" value="360" />
	  <param name="pixel_format" value="mjpeg" />
	  <param name="camera_frame_id" value="front_cam" />
	  <param name="io_method" value="mmap" />
	  <param name="framerate" value="30" />
	  <param name="autoexposure" value="false" />
<!--	  <param name="camera_info_url" value="file://$(find env_percept)/config/front_cam.yaml" />-->
<!--	  <param name="camera_name" value="front_cam" />-->
	</node>


	<node if="$(arg down_cam_enable)" pkg="image_proc" type="image_proc" name="down_image_proc" output="screen" ns="down_cam" />

</launch>
